src/constraints.jl

```diff
@@ -971,9 +971,11 @@ julia> list_of_constraint_types(model)
  (VariableRef, MathOptInterface.ZeroOne)
 ```
 """
-function list_of_constraint_types(model::Model)
-    list = MOI.get(
-        model, MOI.ListOfConstraints())::Vector{Tuple{DataType, DataType}}
-    return Tuple{DataType, DataType}[(jump_function_type(model, f), s)
-                                     for (f,s) in list]
+function list_of_constraint_types(model::Model)::Vector{Tuple{DataType, DataType}}
```

Comment:
Why adding the annotation here as we already annotate the element type of the vector

Reply:
See the comment below. It didn't infer correctly.